clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -145,46 +154,72 @@ private void readHeader() throws IOException {
      * @throws IOException
      */
     private void readBlock() throws IOException {
-        int blockSize = ByteUtils.readUnsignedIntLE(in);
+        if (in.remaining() < 4) {
+            throw new IOException(PREMATURE_EOS);
+        }
+
+        int blockSize = in.getInt();
```

Comment:
from reading the code, `ByteUtils.readUnsignedIntLE` is a misnomer and should probably be just be called `readIntLE`. Unlike `readUnsignedInt`, which returns a long and applies the proper bit mask, `readUnsignedIntLE` returns an integer and doesn't do anything other than shifting the bytes.